lib/ansible/modules/cloud/amazon/lambda.py

```diff
@@ -298,17 +300,18 @@ def main():
     except (botocore.exceptions.ClientError, botocore.exceptions.ValidationError) as e:
         module.fail_json(msg=str(e))
 
-    if role.startswith('arn:aws:iam'):
-        role_arn = role
-    else:
-        # get account ID and assemble ARN
-        try:
-            iam_client = boto3_conn(module, conn_type='client', resource='iam',
-                                region=region, endpoint=ec2_url, **aws_connect_kwargs)
-            account_id = iam_client.get_user()['User']['Arn'].split(':')[4]
-            role_arn = 'arn:aws:iam::{0}:role/{1}'.format(account_id, role)
-        except (botocore.exceptions.ClientError, botocore.exceptions.ValidationError) as e:
-            module.fail_json(msg=str(e))
+    if state == 'present':
+        if role.startswith('arn:aws:iam'):
+            role_arn = role
+        else:
+            # get account ID and assemble ARN
+            try:
+                iam_client = boto3_conn(module, conn_type='client', resource='iam',
+                                    region=region, endpoint=ec2_url, **aws_connect_kwargs)
+                account_id = iam_client.get_user()['User']['Arn'].split(':')[4]
+                role_arn = 'arn:aws:iam::{0}:role/{1}'.format(account_id, role)
+            except (botocore.exceptions.ClientError, botocore.exceptions.ValidationError) as e:
+                module.fail_json(msg=str(e))
```

Comment:
so, the "standard exception handling" is kind of the ideal, but with the problem that sometimes, unpredictably, the "request" parameter isn't set.  So I have instead added my fail_json_aws as used in aws_api_gateway and substituted that in for most cases.  This should really be refactored into module_utils, but the patch for that will not be ready soon so it's easiest to start with a local copy and then refactor later.